### DISCUSSION TITLE
How to add validation on required schema field

### DISCUSSION THREAD

**Discussion by elmalakomar** [davidebianchi/gswagger#106] [Q&A]
Hi,
I'm having problems trying to add a validation on a json schema header. The desired behavior is to have an error if in the Headers of he request it's missing the field "Authorization" or the field value is empty.

I tried two approaches but I seems the validation is not doing the expected behavior:
1)
```
swagger.Definitions{
  Headers: swagger.ParameterValue{
    "Authorization": {
      Schema: &swagger.Schema{
        Value: "",
      },
      Description: "JWT access token",
    },
  },
}
```

2)
```
swagger.Definitions{
  Headers: swagger.ParameterValue{
    "Authorization": {
      Schema: &swagger.Schema{
        Value: &AuthorizationHeader{},,
      },
      Description: "JWT access token",
    },
  },
}
```
with AuthorizationHeader defined like this:
```
type AuthorizationHeader struct {
  Authorization string `json:"Authorization" validate:"required"`
}
```

How can I achieve the expected behavior? Thanks!

Note: The `validate:"required"` annotation comes from this library: https://pkg.go.dev/github.com/go-playground/validator/v10@v10.11.2
I used it to validate a body request field and it works.

**Comment by davidebianchi** [comment]
Hi @elmalakomar! How do you do the validation with the library? I think you could use directly the `AuthorizationHeader` struct of the second example to validate it.

I think it's an interesting use case, which could be also integrated optionally inside gswagger (so, the validation of the request which if not as desired returns 400). What do you think about?

**Reply by elmalakomar** [reply]
Hi @davidebianchi , 
the library let you instantiate a validator that do a check between the variable you are passing and how the related struct is defined.
In the struct you declare some validation tags for each field.
This is a simple example supplied by the library: https://raw.githubusercontent.com/go-playground/validator/master/_examples/simple/main.go

In a project what I did is to define a middleware that checks the input before start the handler function. I wanted to replicate something similar to the custom-plugin-lib.

**Reply by davidebianchi** [reply]
Ok, clear. The validation on request is made by fastify in the custom-plugin-lib example.
How have you made the validation of the headers?
